Example mod/Mod.cs

```diff
@@ -1,123 +1,95 @@
+
+    [Menu("SMLHelper Example Mod")]
     public class Config : ConfigFile
     {
-        public int ChoiceIndex { get; internal set; }
-        public KeyCode KeybindKey { get; internal set; }
-        public float SliderValue { get; internal set; }
-        public bool ToggleValue { get; internal set; }
-    }
+        [Label("My index-based choice"), Tooltip("A simple tooltip"), Choice("Choice 1", "Choice 2", "Choice 3"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public int ChoiceIndex;
 
-    public class Options : ModOptions
-    {
-        public Options() : base("SMLHelper Example Mod")
-        {
-            ChoiceChanged += Options_ChoiceChanged;
-            KeybindChanged += Options_KeybindChanged;
-            SliderChanged += Options_SliderChanged;
-            ToggleChanged += Options_ToggleChanged;
+        [Label("My string-based choice"), Choice("Foo", "Bar"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public string ChoiceString = "Foo";
 
-            GameObjectCreated += Options_GameObjectCreated;
-        }
+        [Label("My enum-based choice"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public CustomChoice ChoiceEnum;
 
-        public void Options_ChoiceChanged(object sender, ChoiceChangedEventArgs e)
-        {
-            if (e.Id != "exampleChoice") return;
-            ExampleMod.Config.ChoiceIndex = e.Index;
-            ExampleMod.Config.Save();
-        }
-        public void Options_KeybindChanged(object sender, KeybindChangedEventArgs e)
-        {
-            if (e.Id != "exampleKeybind") return;
-            ExampleMod.Config.KeybindKey = e.Key;
-            ExampleMod.Config.Save();
-        }
-        public void Options_SliderChanged(object sender, SliderChangedEventArgs e)
-        {
-            if (e.Id != "exampleSlider") return;
-            ExampleMod.Config.SliderValue = e.Value;
-            ExampleMod.Config.Save();
-        }
-        public void Options_ToggleChanged(object sender, ToggleChangedEventArgs e)
-        {
-            if (e.Id != "exampleToggle") return;
-            ExampleMod.Config.ToggleValue = e.Value;
-            ExampleMod.Config.Save();
-        }
+        [Label("My customised enum-based choice"), Choice("1", "2", "3"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public CustomChoice ChoiceEnumCustomValues;
+
+        [Label("My keybind"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public KeyCode KeybindKey;
 
-        public override void BuildModOptions()
+        [Label("My slider"), Slider(0, 50, DefaultValue = 25, Format = "{0:F2}"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public int SliderValue;
+
+        [Label("My stepped slider"), Slider(0, 100, Step = 10), OnChange(nameof(MyGenericValueChangedEvent))]
+        public int SteppedSliderValue;
+
+        [Label("My checkbox"), OnChange(nameof(MyCheckboxToggleEvent)), OnChange(nameof(MyGenericValueChangedEvent))]
```

Comment:
For the sake of consistency, shouldn't there be a `Toggle` attribute here before the `OnChange` attribute?  
There is already an attribute for `Slider` and `Choice`.  
It likely isn't strictly necessary for it to function on a `bool` option but having a `Toggle` attribute would continue the pattern established above.

```CS
[Label("My checkbox"), Toggle, OnChange(nameof(MyCheckboxToggleEvent)), OnChange(nameof(MyGenericValueChangedEvent))]
```

Reply:
> **NOTE:** There is no `KeybindAttribute` or `ToggleAttribute` as these are merely inferred by the field/property type:
`KeyCode` -> `ModKeybindOption`, `bool` -> `ModToggleOption`

I don't mind adding them if we really want them, it just seems like unnecessary code.

Reply:
Bc. if you are not ignoring unattributed members, then any public bool will be picked up as a toggle. If you are ignoring unattributed methods, just giving it a label will pick it up as a toggle, etc. Again, I can add them, it just seems like a waste of bytes.

Reply:
Resolving to maintain a single discussion  https://github.com/SubnauticaModding/SMLHelper/pull/173#discussion_r463896532